frontend/main.go

```diff
@@ -145,16 +146,24 @@ func main() {
 		log.Fatalf("Error creating Kubernetes client: %s", err)
 	}
 
-	// Start server
-	r, err := startHTTPServer(&flags)
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
-	}
-
-	log.Println("Starting Odigos UI...")
-	log.Printf("Odigos UI is available at: http://%s:%d", flags.Address, flags.Port)
-	err = r.Run(fmt.Sprintf("%s:%d", flags.Address, flags.Port))
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
+	// Attempt to start the server on port 3000 and increment the port number if it's already in use
+	for {
+		r, err := startHTTPServer(&flags)
+		if err != nil {
+			log.Printf("Error starting server on port %d: %s", flags.Port, err)
+			flags.Port++ // Try the next port
+		} else {
+			log.Printf("Starting Odigos UI...")
+			log.Printf("Odigos UI is available at: http://%s:%d", flags.Address, flags.Port)
+			err = r.Run(fmt.Sprintf("%s:%d", flags.Address, flags.Port))
+			if err != nil {
+				log.Printf("Error starting server on port %d: %s", flags.Port, err)
+				flags.Port++ // Try the next port
```

Comment:
I see we check for errors in 2 places: `startHTTPServer(&flags)` and `r.Run(fmt.Sprintf("%s:%d", flags.Address, flags.Port))`, but I guess only one of them returns the err when the port is already in use. We might simplify the flow and attempt a new port only in the path where it is expected, and fail fast if the error is unrelated to the port.

Reply:
You are correct; only the r.Run call is likely to return an error related to the port being already in use.
To simplify the flow and fail fast when encountering errors unrelated to the port, i moved the port increment logic to the part of the code where r.Run is called. This way, it will only attempt to increment the port when it know it's necessary.